lib/ConnectionConfig.js

```diff
@@ -106,14 +108,19 @@ ConnectionConfig.getDefaultFlags = function getDefaultFlags(options) {
     '+LONG_PASSWORD',     // Use the improved version of Old Password Authentication
     '+MULTI_RESULTS',     // Can handle multiple resultsets for COM_QUERY
     '+ODBC',              // Special handling of ODBC behaviour
-    '-PLUGIN_AUTH',       // Does *NOT* support auth plugins
+    '+PLUGIN_AUTH',       // Supports auth plugins
     '+PROTOCOL_41',       // Uses the 4.1 protocol
     '+PS_MULTI_RESULTS',  // Can handle multiple resultsets for COM_STMT_EXECUTE
     '+RESERVED',          // Unused
     '+SECURE_CONNECTION', // Supports Authentication::Native41
     '+TRANSACTIONS'       // Expects status flags
   ];
 
+  if (options && options.database) {
```

Comment:
Is this change related specifically to the MySQL 8 support, or just a general fix. Just curious.

Reply:
For the new authentication magic to happen in the server, we need to explicitly provide the authentication plugin name in the `ClientAuthenticationPacket`, and, doing that, we need to be careful of which flags to enable, otherwise the server will, for instance, interpret `caching_sha2_password` or `mysql_native_password` as the schema/database name, if none is provided.

I didn't want to mess around with the `PacketWriter` and this seemed the easiest way to overcome that hurdle. However, we can try to think of something else if it feels weird.

Reply:
Sure, but I was more curious on if this is a general fix or specific to MySQL 8. But even then, users can pass in their own `flags` list, and this module will enable / disable them unless it's listed here with a flag to force one way or the other. If having it set when it shouldn't is an issue, then it should probably be flagged here so the users cannot pass in the flag and get it set when no db is specified.